distsql/select_result.go

```diff
@@ -167,6 +172,22 @@ func (r *selectResult) getSelectResp() error {
 	}
 }
 
+func (r *selectResult) updateCopRuntimeStats(callee string) {
+	if r.ctx.GetSessionVars().StmtCtx.RuntimeStatsColl == nil ||
+		callee == "" || len(r.selectResp.GetExecutionSummaries()) != len(r.copPlanIDs) {
+		return
+	}
+
+	for i, detail := range r.selectResp.GetExecutionSummaries() {
+		if detail != nil && detail.TimeProcessedNs != nil &&
```

Comment:
when will TimeProcessedNs or NumProducedRows or NumIterations be nil?

Reply:
Just a protect?

Reply:
Yes, just a protection to prevent from panic caused by wrong result from TiKV.